internal/scheduler/scheduler.go

```diff
@@ -228,23 +228,25 @@ func (s *Scheduler) runJob(ctx context.Context, r *job.Run) error {
 	jobContext, rj.cancelCtx = context.WithCancel(ctx)
 
 	go func() {
-		defer rj.cancelCtx()
 		runErr := j.Run(jobContext)
-		var updateErr error
+		// Job has ended, cancel job run specific context to clean up
+		rj.cancelCtx()
 
 		// Get final status report to update run progress with
 		status := j.Status()
-
-		switch runErr {
-		case nil:
+		var updateErr error
+		switch {
+		case ctx.Err() != nil:
+			// Base context is no longer valid, skip repo updates as they will fail and exit
+		case runErr == nil:
 			nextRun, inner := j.NextRunIn()
 			if inner != nil {
 				event.WriteError(ctx, op, inner, event.WithInfoMsg("error getting next run time", "name", j.Name()))
 			}
-			_, updateErr = repo.CompleteRun(jobContext, r.PrivateId, nextRun, status.Completed, status.Total)
+			_, updateErr = repo.CompleteRun(ctx, r.PrivateId, nextRun, status.Completed, status.Total)
```

Comment:
If `inner != nil` what will the value of `nextRun` be when this is called?

Reply:
Depends on the job implementation https://github.com/hashicorp/boundary/blob/main/internal/scheduler/job.go#L20